Add `Beatmap/Pattern/osu!mania` listing and subpages (#8595)
ppy/osu-wiki: wiki/Beatmap/Pattern/osu!mania/Stream/en.md

---
stub: true
---

# Stream

*For specific stream types unrelated to 4-key, refer to [Chordstream](/wiki/Beatmap/Pattern/osu!mania/Chordstream).*

> **Streams** are constant notes within the same snap interval that occur for at least one beat, avoiding an occurance of a [minijack](/wiki/Beatmap/Pattern/osu!mania/Jack#minijack). There are many different types of streams that impose different levels of strain on the player. When a stream is used on its own, it's likely referring to a **single note stream**.

Reviewer: many snap intervals can fit in one beat, so the extra remark doesn't make sense

```suggestion
A **stream** is a constant set of notes with the same snap interval. There are many different types of streams that impose different levels of strain on the player. The term *stream* on its own usually refers to a **single-note stream**.
```

![](img/stream.png "This image shows a specific type of stream known as a roll.")

> A **roll** is a common variation of a stream consisting of notes that go from the first to the last column or last to the first column.

Reviewer: ```suggestion
A **roll** is a common variation of a stream consisting of notes that go across all columns in sequence.
```

> **Streams** that appear shorter than one beat are typically referred to as **bursts**, which are a shorter and quick succession of notes. They commonly use snaps higher than 1/4th.

Reviewer: ```suggestion
Streams with notes appearing in intervals faster than one beat are typically referred to as **bursts**. They commonly use snaps higher than 1/4.
```

![](img/burst.png)

> ## Jumpstream

Reviewer: if all of the below are types of chordstreams, then why aren't they just put into the chordstream article? or rather, why is "chordstream" a separate article, when it is a type of stream?

Author: I debated this before but I think they should be separated due to the term chordstream never being used in 4k and instead terms like jumpstream and handstream (omitting quadstream since it's a 4k exclusive term) are used, whilst within 7k this term isn't used as much and instead chordstreams are favoured. This is mostly because of how much of an impact the difference in densities can make on patterning in 4k due to it's limited columns but also since 7k is much more likely to go beyond chordsizes of 3 more frequently. 

The wiki is also _kind of_ moreso centred around 4k patterns since it's the most popular mode by quite a big margin, so I though it would make sense to keep the terms exclusive to 7k separated to avoid confusion along with give people who are looking for that specific terminology a more optimal place to reference back to instead of one long article page in which half of the information is practically irrelevant but technically still applies.

Reviewer: i disagree with separating 4k and 7k terms

- if anything, wouldn't it be "4k" and "other key modes"? but that makes even less sense and way too arbitrary. there's no kind of categorisation based on popularity anywhere else on the wiki, as that would be biased
- it's not like 4k terms aren't used in 7k and other key modes?
- it would be better to just explain which terms are used in what contexts

Reviewer: i disagree with separating 4k and 7k terms

- if anything, wouldn't it be "4k" and "other key modes"? but that makes even less sense and way too arbitrary. there's no kind of categorisation based on popularity anywhere else on the wiki, as that would be biased
- it's not like 4k terms aren't used in 7k and other key modes?
- it would be better to just explain which terms are used in what contexts

Author: "Other keymodes" was already initially referred to though I see it was changed already, it doesn't really seem that arbitrary to me considering that is how osu!mania is mainly divided (i.e. global leaderboards being split into 4k/7k rankings, RC having exclusive guidelines for both 4k and 7k, etc.). This is simply the basis for how osu!mania has operated for years already because they keymodes are so different and 4k is statistically the more populated mode by a big margin.

**In the case of streams** (important because these are some of the only terms like this), 4k terms like the aforementioned are not really, if not never used in 7k. Again, terms like jumpstream, handstream, and chordstream stem from different keymodes for different reasons entirely and so their meaning is not necessarily building on each other but rather distinct, and so it's not easily interchangable. So, although their literal definitions are technically applicable, they just don't make any sense to practically use among the 7k community.

I really think it would just make more sense to keep them separated to avoid all the technicalities and overlap altogether, having the chordstream page just mention that these terms are exclusive to modes outside of 4k would align moreso with how these definitions function practically.

Reviewer: then let's keep them separate but have them in the same article like this. what do you think?

having a separate adjacent article for a subtype is just too much jank

```markdown
# Stream

## Single-note stream

General term

## Jumpstream

4K term

## Handstream

4K term

## Quadstream

4K term

## Chordstream

Non-4K term
```

Author: Sure, seems reasonable. I applied it now so let me know if there's anything that should be fixed

> **Jumpstreams** are streams that involve [jumps](/wiki/Beatmap/Pattern/osu!mania/Chord#jump). Usually, the stream itself is on 1/4th snaps whilst the jumps land on snaps greater than 1/4th. This is the most common type of stream pattern used in 4-key osu!mania.

Reviewer: ```suggestion
**Jumpstreams** are streams that involve [jumps](/wiki/Beatmap/Pattern/osu!mania/Chord#jump). Usually, the stream itself is on 1/4 snaps while the jumps land on higher snaps. This is the most common type of stream pattern used in 4K osu!mania.
```

![](img/jumpstream.png)

 It's also loosely referred to as a type of chordstream, which mixes different types of chords and streams.

## Handstream

> **Handstreams** are streams that involve [hands](/wiki/Beatmap/Pattern/osu!mania/Chord#hand), and may involve jumps as well. Similar to jumpstreams, the stream itself is on 1/4th snaps whilst the chords can land on snaps greater than 1/4th.

Reviewer: ```suggestion
**Handstreams** are streams that involve [hands](/wiki/Beatmap/Pattern/osu!mania/Chord#hand), and may involve jumps as well. Like jumpstreams, the stream itself is on 1/4 snaps while the jumps land on higher snaps.
```

![](img/handstream.png)

This is also a type of chordstream.

## Quadstream

> **Quadstreams** is a term that is almost exclusively used for 4K osu!mania. Similar to the previous two types of streams, it's a stream that involves the usage of quads as well as other smaller chords. Contrastingly, it creates [minijacks](/wiki/Beatmap/Pattern/osu!mania/Jack#minijack) between the quad and the succession of notes.

Reviewer: ```suggestion
**Quadstream** is a term almost exclusively used in 4K osu!mania. Similar to the previous two types of streams, it is a stream that involves the usage of quads as well as other smaller chords. In 4K osu!mania, the quads in quadstreams invariably create [minijacks](/wiki/Beatmap/Pattern/osu!mania/Jack#minijack) with the note after the quad.
```

Author: I'd be inclined to slightly alter the change to the last sentence since it can create a minijack before the quad as well, which is why I choose to use "between the quad and the succession of notes" instead. I think maybe this would work better without cluttering the sentence too much:
```suggestion
**Quadstream** is a term almost exclusively used in 4K osu!mania. Similar to the previous two types of streams, it is a stream that involves the usage of quads as well as other smaller chords. In 4K osu!mania, the quads in quadstreams create [minijacks](/wiki/Beatmap/Pattern/osu!mania/Jack#minijack) with the note before or after the quad.
```
